plugins/csp-trajectories/src/DeepSpaceDot.cpp

```diff
@@ -158,21 +176,24 @@ bool DeepSpaceDot::Do() {
   glGetFloatv(GL_PROJECTION_MATRIX, glMatP.data());
   auto matMV = glm::make_mat4x4(glMatMV.data()) * glm::mat4(object->getObserverRelativeTransform());
 
+  glPushAttrib(GL_ENABLE_BIT | GL_COLOR_BUFFER_BIT | GL_DEPTH_BUFFER_BIT);
+
   glEnable(GL_BLEND);
   glDepthMask(GL_FALSE);
+
   glBlendFunc(GL_SRC_ALPHA, GL_ONE_MINUS_SRC_ALPHA);
 
   // draw simple dot
   mShader.Bind();
   glUniformMatrix4fv(mUniforms.modelViewMatrix, 1, GL_FALSE, glm::value_ptr(matMV));
   glUniformMatrix4fv(mUniforms.projectionMatrix, 1, GL_FALSE, glMatP.data());
-  mShader.SetUniform(mUniforms.color, pColor.get()[0], pColor.get()[1], pColor.get()[2]);
-  mShader.SetUniform(mUniforms.aspect, fAspect);
+  mShader.SetUniform(mUniforms.color, pLuminance.get() * pColor.get()[0],
+      pLuminance.get() * pColor.get()[1], pLuminance.get() * pColor.get()[2], pColor.get()[3]);
```

Comment:
If `pColor` would be a glm type it would be much more concise to pass it to the shader.